### PR TITLE
tests: Use a function to build path for utility images

### PR DESCRIPTION
**Pull request by vasiliy-ul**
### What this PR does

The PR introduces a common function to build full registry paths using the provided image name and respecting the utility repo prefix and tag.

Before this PR: image paths were hardcoded in some places without taking into account the `flags` (i.e. repo prefix and tag).

After this PR: common function is used. No more hardcoding.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by victortoso** [comment]
The only other place my grep found `KubeVirtUtility` that might be related to this PR is in [tests/storage/datavolume.go:536](https://github.com/kubevirt/kubevirt/blob/main/tests/storage/datavolume.go#L536), to split name/port.

IMHO, PR is good as is so, /lgtm

**Review by fossedihelm — APPROVED** [review]
Thank you!

File: tests/util/images.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+package util
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/tests/flags"
+)
+
+func GetUtilityImageFromRegistry(imageName string) string {
```
Comment (EdDev) [inline tests/util/images.go:28]: I would prefer not to see additions under any generic `util` package.

Can this be placed in a more focused package?
E.g. `libimageregistry` or `libregistry`.

Comment (vasiliy-ul) [reply tests/util/images.go:28]: @EdDev, what do you think about adding the function to `libinfra` instead? Does it look like a suitable place? I do not have a strong opinion on that, tbh, so I am also fine with creating a new package.

Comment (EdDev) [reply tests/util/images.go:28]: If I only knew what this `libinfra` represents.
Based on git history, it got helpers from a prev test file names `infra_test.go`.

Do you think the image registry is related somehow? I just do not know.

Comment (EdDev) [reply tests/util/images.go:28]: @dhiller , can you please suggest what this `libinfra` represents exactly?

Comment (vasiliy-ul) [reply tests/util/images.go:28]: My guess was that it stand for `infrastructure` stuff. Currently, it includes smth related to certificates, http, node-labeler, etc. Just assumed that utility container images might fall into the infra category as well.

Comment (EdDev) [reply tests/util/images.go:28]: Well, it is better than `utils`.
It does have the potential to collect another version of unrelated stuff.

I'll leave the decision for you, I have no strong feelings about it.

Comment (vasiliy-ul) [reply tests/util/images.go:28]: Tried to move the function to `libinfra`. After importing it where needed, got a circular dependency:
```
ERROR: /root/go/src/kubevirt.io/kubevirt/tests/testsuite/BUILD.bazel:3:11: in go_library rule //tests/testsuite:go_default_library: cycle in dependency graph:
    //cmd/dump:dump (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
    //cmd/dump:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
    //tests/reporter:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
.-> //tests/testsuite:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
|   //tests/libstorage:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
|   //tests/libinfra:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
|   //tests:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
|   //tests/libwait:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
`-- //tests/testsuite:go_default_library (3e1517af6a4f7eb088e2179e4d66b4fd00494be72ad1028a2ff1c0eb967eafe2)
ERROR: Analysis of target '//cmd/dump:dump' failed; build aborted
```
Seems that there might be some mess to take care of before doing that.

Comment (fossedihelm) [reply tests/util/images.go:28]: > Looks like I am not alone with that pain
> 
> https://github.com/kubevirt/kubevirt/blob/9db1b1301b2dbd8261020006b67de1a2663f6cbc/tests/migration/eviction_strategy.go#L257-L261

Here I am :slightly_smiling_face: :raising_hand_man:

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
@EdDev please unhold if you agree.

File: tests/util/images.go (hunk 1 above)
Comment (EdDev) [reply tests/util/images.go:28]: @vasiliy-ul , can you try and place it in the previously suggested new package?

`libimageregistry` or `libregistry` ?

Comment (vasiliy-ul) [reply tests/util/images.go:28]: Well, I tried to fix the cycle. But eventually faced a few more. There is a need to decouple a lot of stuff to fully address the problem, and the refactoring turns out pretty messy at the end. So yeah, probably for now if that is okay, I will go with a new `libregistry` package.

I will take one more look and probably open a follow-up PR with some attempts to move things around in the hope to make things a bit better.

**Review by EdDev — APPROVED** [review]
Thank you!

**Comment by EdDev** [comment]
> /approve /hold @EdDev please unhold if you agree.

/unhold

**Review by EdDev — APPROVED** [review]

**Comment by vasiliy-ul** [comment]
Rebased the PR. No new changes.

**Review by EdDev — APPROVED** [review]

**Pull request merged**